test/tests.ts

```diff
@@ -142,18 +159,25 @@ describe('Resolvers', () => {
 		);
 
 		const bundle = new Resolve.Bundle(stream, '', arch);
-		return Resolve.resolveBundle(bundle, resolvers).then(resolved => {
-			assert(
-				resolved.projectType === name,
-				'Architecture specific dockerfile resolver not used',
-			);
-
-			return getDockerfileFromTarStream(resolved.tarStream).then(contents => {
-				assert(
-					contents.trim() === 'correct',
-					'Incorrect architecture chosen for project resolve',
-				);
+		const outStream = Resolve.resolveInput(bundle, resolvers);
+
+		return new Promise(async (resolve, reject) => {
+			outStream.on('error', reject);
+			outStream.on('resolver', r => {
+				try {
+					expect(r).to.equal(name);
+				} catch (e) {
+					reject(e);
+				}
 			});
+
+			const dockerfile = await getDockerfileFromTarStream(outStream);
+			try {
+				expect(dockerfile.trim()).to.equal('correct');
+			} catch (e) {
+				reject(e);
+			}
+			resolve();
```

Comment:
This is hard to read.
You could define this helper:
```typescript
async function waitForResolver(outStream: Readable): Promise<string> {
    return await new Promise((resolve, reject) => {
        outStream.on('error', reject);
        outStream.on('resolver', resolve);
    });
}
```

and replace your tests with:
```typescript
const r = await waitForResolver(outStream);
expect(r).to.equal(name);
const dockerfile = await getDockerfileFromTarStream(outStream);
expect(dockerfile.trim()).to.equal('correct');
```

and removing these `try`/`catch`es.